Source/WebCore/inspector/agents/InspectorPageAgent.cpp

```diff
@@ -753,7 +760,8 @@ Inspector::Protocol::ErrorStringOr<void> InspectorPageAgent::setCookie(Ref<JSON:
         auto* page = document->page();
         if (!page)
             continue;
-        page->cookieJar().setRawCookie(*document, cookie.value());
+        auto isPartitioned = shouldPartition.value_or(false) ? IsPartitionedCookie::Yes : IsPartitionedCookie::No;
```

Comment:
if the developer modifies a cookie that has an existing `partitionKey` will this preserve that or generate a new one?  would be great to have some tests for this if possible :)

Reply:
It will recompute the key. I suppose this could be surprising behavior because the key that is shown in the popup could be different from the key that is actually set on the cookie. However, that should be impossible, because none of the values that can be modified in WebInspector would influence computed partition key, and we only show cookies in WebInspector that match the original partition key, so everything should match. An alternative is we could rename this enum from `IsPartitionedCookie` to `ShouldPartitionCookie`, and then we only set a new partition key if that is necessary.